src/PreCICE.jl

```diff
@@ -64,227 +85,1031 @@ function createSolverInterfaceWithCommunicator(participantName::String,
     ccall((:precicec_writeBlockVectorData, "libprecice"), Cvoid, (Cint, Cint, Ref{Cint}, Ref{Cdouble}), dataID, size, valueIndices, values)
 end
 
 
+# TODO Are they provided as a block of continuous memory?
+@doc """
+
+    writeVectorData(dataID::Integer, valueIndex::Integer, dataValue::AbstractArray{Float64})
+
+Write vectorial floating-point data to a vertex. This function writes a value of a specified vertex to a dataID.
+Values are provided as a block of continuous memory.
+
+The 2D-format of value is a array of shape 2
+
+The 3D-format of value is a array of shape 3
+
+# Arguments
+- `dataID::Integer`: ID of the data to be written. Obtained by [`getDataID`](@ref).
+- `valueIndex::Integer`: Index of the vertex. 
+- `dataValue::AbstractArray{Float64}`: The array holding the values.
+
+# Notes
+        
+Previous calls:
+ - Count of available elements at `value` matches the configured dimension
+ - [`initialize`](@ref) has been called
+
+Examples:
+
+Write vector data for a 2D problem with 5 vertices:
+```julia
+data_id = 1
+vertex_id = 5
+value = [v5_x, v5_y]
+writeVectorData(data_id, vertex_id, value)
+```
+
+Write vector data for a 3D (D=3) problem with 5 (N=5) vertices:
+```julia
+data_id = 1
+vertex_id = 5
+value = [v5_x, v5_y, v5_z]
+writeVectorData(data_id, vertex_id, value)
+```
+"""
 function writeVectorData(dataID::Integer, valueIndex::Integer, dataValue::AbstractArray{Float64})
     ccall((:precicec_writeVectorData, "libprecice"), Cvoid, (Cint, Cint, Ref{Cdouble}), dataID, valueIndex, dataValue)
 end
 
+# TODO same as above
+@doc """
+
+    writeBlockScalarData(dataID::Integer, size::Integer, valueIndices::AbstractArray{Cint}, values::AbstractArray{Float64})
+
+Write scalar data given as block.
+
+This function writes values of specified vertices to a dataID. Values are provided as a block of continuous memory. `valueIndices` contains the indices of the vertices.
+
+# Arguments
+- `dataID::Integer`: ID of the data to be written. Obtained by getDataID().
+- `size::Integer`: 	Number n of vertices. 
+- `valueIndices::AbstractArray{Cint}`: Indices of the vertices.
+- `values::AbstractArray{Float64}`: The array holding the values.
+    
+# Notes
+
+Previous calls:
+ - Count of available elements at `values` matches the given size
+ - Count of available elements at `vertex_ids` matches the given size
+ - [`initialize`](@ref) has been called
 
+# Examples
+
+Write block scalar data for a 2D and 3D problem with 5 (N=5) vertices:
+```julia
+data_id = 1
+vertex_ids = [1, 2, 3, 4, 5]
+values = [1, 2, 3, 4, 5]
+writeBlockScalarData(data_id, vertex_ids, values)
+```
+"""
 function writeBlockScalarData(dataID::Integer, size::Integer, valueIndices::AbstractArray{Cint}, values::AbstractArray{Float64})
     ccall((:precicec_writeBlockScalarData, "libprecice"), Cvoid, (Cint, Cint, Ref{Cint}, Ref{Cdouble}), dataID, size, valueIndices, values)
 end
 
+# TODO example is strange
```

Comment:
Example looks okay to me. You can also use some dummy value instead of `value = v5`, like `value = 1.0`